packages/python-packages/apiview-copilot/src/_github_manager.py

```diff
@@ -0,0 +1,273 @@
+        """
+        url = f"https://api.github.com/repos/{owner}/{repo}/issues/{issue_number}/labels"
+        use_app = self.use_app_auth if use_app_auth is None else use_app_auth
+        token = self._get_installation_token(owner) if use_app else self._pat
+        return self._request_json("POST", url, token, json={"labels": list(labels)})
+
+    def close_issue(self, *, owner: str, repo: str, issue_number: int, use_app_auth: Optional[bool] = None) -> dict:
+        """
+        Closes a GitHub issue.
+        """
+        url = f"https://api.github.com/repos/{owner}/{repo}/issues/{issue_number}"
+        use_app = self.use_app_auth if use_app_auth is None else use_app_auth
+        token = self._get_installation_token(owner) if use_app else self._pat
+        return self._request_json("PATCH", url, token, json={"state": "closed"})
+
+    def _request_json(
+        self,
+        method: str,
+        url: str,
+        token: str,
+        *,
+        json: Optional[dict] = None,
+        as_app: bool = False,  # ignored for PAT
+    ) -> dict | list:
+        headers = {"Authorization": f"token {token}"}
+        last_exc = None
+        for attempt in range(1, self._max_retries + 1):
+            try:
+                resp = self._client.request(method, url, headers=headers, json=json)
+                if resp.status_code in (429, 502, 503, 504) or (
+                    resp.status_code == 403 and "secondary rate limit" in resp.text.lower()
+                ):
+                    raise httpx.HTTPStatusError("Retryable", request=resp.request, response=resp)
+                resp.raise_for_status()
+                if resp.text:
+                    return resp.json()
+                return {}
+            except httpx.HTTPError as exc:
+                last_exc = exc
+                logger.warning(
+                    "GitHub request failed (%s %s) attempt %d/%d: %s", method, url, attempt, self._max_retries, exc
+                )
+                if attempt == self._max_retries:
+                    break
+                time.sleep(self._backoff_s * attempt)
+        if isinstance(last_exc, httpx.HTTPStatusError) and last_exc.response is not None:
+            try:
+                detail = last_exc.response.json()
+            except Exception:
+                detail = last_exc.response.text
+            raise RuntimeError(f"GitHub API error {last_exc.response.status_code}: {detail}") from last_exc
+        raise last_exc or RuntimeError("Unknown GitHub request failure")
+
+    # === GitHub App authentication ===
+
+    def _create_app_jwt(self) -> str:
+        """
+        Mint a GitHub App JWT using Azure Key Vault 'sign' (non-exportable key).
+        """
+        settings = SettingsManager()
+        keyvault_url = settings.get("github_app_keyvault_url")
+        key_name = settings.get("github_app_key_name")
+        app_id = settings.get("github_app_id")
+
+        credential = DefaultAzureCredential()
```

Comment:
Do you want to use DAC here? I would suggest a more specific set of credentials.

Reply:
I updated it to use our `get_credential` method, but under the hood this is just DefaultAzureCredential. What more specific set of credentials do you suggest?

Reply:
In general you should prefer a specific chained credential and only include the credential types that you expect to use. The biggest issue with DAC is that it probes too many options and you often end up using a credential you don't want to use.